src/SourceBuild/content/eng/tools/ExclusionsLibrary/Exclusion.cs

```diff
@@ -0,0 +1,72 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Extensions.FileSystemGlobbing;
+
+namespace ExclusionsLibrary;
+
+internal class Exclusion : IEquatable<Exclusion>
+{
+    /// <summary>
+    /// The pattern to match.
+    /// </summary>
+    public string Pattern { get; init; }
+
+    /// <summary>
+    /// The suffixes to match.
+    /// </summary>
+    public HashSet<string?> Suffixes { get; init; }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="Exclusion"/> class.
+    /// <param name="line"/>>The line to parse.</param>
+    /// </summary>
+    public Exclusion(string line)
```

Comment:
Rather than a constructor, my preference is to have a static `Parse` method instead as it better indicates what is happening and aligns with other .NET APIs. If you then also changed the constructor signature to `(string pattern, Hash<string?> suffixes)`, which the `Parse` method would call, then you could also get rid of the `Exclusion(Exclusion other)` constructor.